File: app/lib/models/exercise.dart
class Exercise {
  final int id;
  final String name;
  final String category;
  final String? description;
  final String? technique;
  final List<dynamic> muscleGroups;
  final int difficulty;

  Exercise({
    required this.id,
    required this.name,
    required this.category,
    this.description,
    this.technique,
    this.muscleGroups = const [],
    this.difficulty = 1,
  });

  factory Exercise.fromJson(Map<String, dynamic> json) => Exercise(
        id: json['id'],
        name: json['name'],
        category: json['category'] ?? 'pesas',
        description: json['description'],
        technique: json['technique'],
        muscleGroups: json['muscle_groups'] ?? [],
        difficulty: json['difficulty'] ?? 1,
      );
}

File: backend/app/Models/Exercise.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Exercise extends Model
{
    protected $fillable = [
        'created_by', 'name', 'category', 'description', 'technique',
        'media_url', 'muscle_groups', 'difficulty', 'variations',
    ];

    protected $casts = [
        'muscle_groups' => 'array',
        'variations' => 'array',
    ];

    public function creator(): BelongsTo
    {
        return $this->belongsTo(User::class, 'created_by');
    }
}
